coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -594,6 +594,11 @@ func (s *server) acquireProtoJob(ctx context.Context, job database.ProvisionerJo
 			})
 		}
 
+		ownerRbacRoles := []string{}
+		for _, role := range owner.RBACRoles {
+			ownerRbacRoles = append(ownerRbacRoles, role)
+		}
```

Comment:
Unfortunately this is not sufficient.

This is the code we use to fetch all the user's roles from the database:

https://github.com/coder/coder/blob/main/coderd/httpmw/apikey.go#L447-L475

---

So we should change this to:
```suggestion
	roles, err := s.Database.GetAuthorizationUserRoles(dbauthz.AsSystemRestricted(ctx), owner.ID)
		if err != nil {
			// Should we fail the job?
		}

		ownerRbacRoles := []string{}
		for _, role := range roles.Roles {
			ownerRbacRoles = append(ownerRbacRoles, role)
		}
```

This will include all org roles as `<org-id>:<org-role>`. Would it be possible to have the `role` datatype be a struct, instead of `list(string)`?
```hcl
role {
  name string
  org-id string
}
```

This would be preferred, as roles are uniquely identified by the pair of `(org-id, rolename)`. And roles at the site wide level just have the uuid of `000...000`

Reply:
Makes perfect sense to me! Just updated the PR